src/geyser/bin.zig

```diff
@@ -0,0 +1,278 @@
+                },
+            },
+        },
+    },
+} };
+
+pub fn main() !void {
+    try cli.run(&cli_app, std.heap.page_allocator);
+}
+
+pub fn getOwnerFilters(allocator: std.mem.Allocator) !?std.AutoArrayHashMap(sig.core.Pubkey, void) {
+    const owner_accounts_str = config.owner_accounts;
+    if (owner_accounts_str.len == 0) {
+        return null;
+    }
+
+    var owner_pubkeys = std.AutoArrayHashMap(sig.core.Pubkey, void).init(allocator);
+    try owner_pubkeys.ensureTotalCapacity(@intCast(owner_accounts_str.len));
+    for (owner_accounts_str) |owner_str| {
+        const owner_pubkey = try sig.core.Pubkey.fromString(owner_str);
+        owner_pubkeys.putAssumeCapacity(owner_pubkey, {});
+    }
+
+    return owner_pubkeys;
+}
+
+pub fn csvDump() !void {
+    const allocator = std.heap.c_allocator;
+
+    const metrics_thread = try sig.prometheus.spawnMetrics(allocator, 12355);
+    metrics_thread.detach();
+    std.debug.print("spawing metrics thread on port 12355\n", .{});
+
+    const pipe_path = config.pipe_path;
+    std.debug.print("using pipe path: {s}\n", .{pipe_path});
+
+    // owner filters
+    var maybe_owner_pubkeys = try getOwnerFilters(allocator);
+    defer if (maybe_owner_pubkeys) |*owners| owners.deinit();
+    if (maybe_owner_pubkeys) |owner_pubkeys| {
+        std.debug.print("owner filters: {s}\n", .{owner_pubkeys.keys()});
+    } else {
+        std.debug.print("owner filters: none\n", .{});
+    }
+
+    // csv file to dump to
+    const dump_csv_path = try std.fmt.allocPrint(
+        allocator,
+        "{s}{s}",
+        .{ sig.VALIDATOR_DIR, "accounts.csv" },
+    );
+    defer allocator.free(dump_csv_path);
+
+    const csv_file = try std.fs.cwd().createFile(dump_csv_path, .{});
+    defer csv_file.close();
+
+    std.debug.print("dumping to csv: {s}\n", .{dump_csv_path});
+
+    // setup reader
+    const exit = try allocator.create(std.atomic.Value(bool));
+    defer allocator.destroy(exit);
+    exit.* = std.atomic.Value(bool).init(false);
+
+    var reader = try sig.geyser.GeyserReader.init(allocator, pipe_path, exit, .{
+        .bincode_buf_len = config.geyser_bincode_buf_len,
+        .io_buf_len = config.geyser_io_buf_len,
+    });
+    defer reader.deinit();
+
+    // preallocate memory for csv rows
+    const recycle_fba = try allocator.create(sig.utils.allocators.RecycleFBA(.{ .thread_safe = true }));
+    recycle_fba.* = try sig.utils.allocators.RecycleFBA(.{ .thread_safe = true }).init(allocator, 1 << 32);
+    defer {
+        recycle_fba.deinit();
+        allocator.destroy(recycle_fba);
+    }
+
+    // setup thread to write to csv
+    var io_channel = try sig.sync.Channel([]u8).create(allocator);
+    defer {
+        io_channel.deinit();
+        allocator.destroy(io_channel);
+    }
+
+    const io_handle = try std.Thread.spawn(.{}, csvDumpIOWriter, .{ exit, csv_file, io_channel, recycle_fba });
+    defer io_handle.join();
+    errdefer {
+        exit.store(true, .release);
+    }
```

Comment:
Any reason for the stylistic decision to write this as a block instead of as `errdefer exit.store(true, .release)`? It looks a bit glaring when it's right underneath `defer io_handle.join();`.

Reply:
5b03899